backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java

```diff
@@ -551,4 +564,30 @@ private static void checkSystemEndpoint(EndpointType endpointType) {
     private boolean isEndpointTypeAllowed(EndpointType endpointType) {
         return !featureFlipper.isEmailsOnlyMode() || endpointType.isSystemEndpointType;
     }
+
+    /**
+     * Removes the secrets from the endpoint's properties when returning them
+     * to the client.
+     * @param endpoint the endpoint to redact the secrets from.
+     */
+    @Deprecated(forRemoval = true)
+    protected void redactSecretsForEndpoint(final Endpoint endpoint) {
+        if (endpoint.getProperties() instanceof SourcesSecretable sourcesSecretable) {
+            final BasicAuthentication basicAuthentication = sourcesSecretable.getBasicAuthentication();
+            if (basicAuthentication != null) {
+                basicAuthentication.setPassword("[REDACTED CREDENTIAL]");
```

Comment:
Returning string with `[ ]` looks like some kind of missing placeholder transformation to me. On this case I would prefer sending back sting like `********`, because it's a standard behavior to hide secrets.
But that's my personnal choice :), WDYT ?

Reply:
You are right, yes, the asterisks are kind of universal for redacted secrets.